### PR TITLE
Metadata reader: Preserve type when reading enum values in attributes

### PR DESCRIPTION
**Pull request by ChrisDenton**
Also implement parsing enums in named arguments.

I've gone back on forth on whether to use an `i32`, `u64` or an `Integer` type to store the value. I eventually landed on the latter but please stop me if I'm over-engineering this.

Closes: #2330

### CODE REVIEW

File: crates/libs/metadata/src/reader/mod.rs [hunk 1]
```diff
@@ -127,6 +127,18 @@ pub enum Value {
     F64(f64),
     String(String),
     TypeDef(TypeDef),
+    Enum(TypeDef, Integer),
+}
+
+pub enum Integer {
```
Comment (kennykerr) [inline crates/libs/metadata/src/reader/mod.rs:133]: Looks good.

File: crates/libs/metadata/src/reader/mod.rs [hunk 2]
```diff
@@ -312,14 +314,19 @@ impl<'a> Reader<'a> {
...
-            let name = values.read_str().to_string();
+            let mut name = values.read_str().to_string();
             let arg = match arg_type {
                 0x02 => Value::Bool(values.read_bool()),
                 0x06 => Value::I16(values.read_i16()),
                 0x08 => Value::I32(values.read_i32()),
                 0x09 => Value::U32(values.read_u32()),
                 0x0E => Value::String(values.read_str().to_string()),
                 0x50 => Value::TypeDef(self.get(TypeName::parse(values.read_str())).next().expect("Type not found")),
+                0x55 => {
```
Comment (kennykerr) [inline crates/libs/metadata/src/reader/mod.rs:325]: Is this being tested anywhere? Perhaps add a test to `crates/tests/metadata` that reads the `SupportedArchitecture` attributes on `SetWindowLongPtrA`.

Comment (ChrisDenton) [reply crates/libs/metadata/src/reader/mod.rs:325]: Oh good point, I'll add that. It is being tested in my own code at least ;)

Comment (ChrisDenton) [reply crates/libs/metadata/src/reader/mod.rs:325]: So the only attribute that actually hits this code path is `GCPressure` (which I guess is useful to GCs). I've added a test for it but I don't know how reliable that attribute is.

Comment (kennykerr) [reply crates/libs/metadata/src/reader/mod.rs:325]: As soon as I have the metadata generation completed, we can stop digging around for test cases and just write discrete tests from scratch.

**Review by kennykerr — APPROVED** [review]
Thanks!

**Pull request merged**